changes/probe.py

```diff
@@ -58,11 +68,20 @@ def has_changelog():
 
 
 def has_readme():
-    """README.md"""
+    """README"""
+    readme = None
+    message = 'Create a README file'
+    for filename in os.listdir("."):
+        readme = re.match(r"README($|\.\w{2,})", filename, re.I)
```

Comment:
@goldsborough thanks very much for the detailed analysis! :+1: 
I think I like the clarity of intent of the `any` version (and probably would even if it wasn't the most efficient, since we're primarily running on development machines).
I'm ok with the generic error message (in conjunction with appropriate documentation in the `changes` README)